templates/route-RedHat.erb

```diff
@@ -2,5 +2,5 @@
 ### File managed by Puppet
 ###
 <%- (0..(@ipaddress.length-1)).each do |id| -%>
-<%= @ipaddress[id] %>/<%= @netmask[id] %><%- if @gateway and @gateway[id] -%> via <%= @gateway[id] %><%- end -%> dev <%= @interface %><%- if @scope and @scope[id] -%> scope <%= @scope[id] %><%- end -%><%- if @source and @source[id] -%> src <%= @source[id] %><%- end -%><%- if @table and @table[id] -%> table <%= @table[id] %><% end %><%- if @metric and @metric[id] -%> metric <%= @metric[id] %><% end %>
+<%- if @family and @family[id] != 'inet6' -%><%= @ipaddress[id] %>/<%= @netmask[id] %><%- if @gateway and @gateway[id] -%> via <%= @gateway[id] %><%- end -%> dev <%= @interface %><%- if @scope and @scope[id] -%> scope <%= @scope[id] %><%- end -%><%- if @source and @source[id] -%> src <%= @source[id] %><%- end -%><%- if @table and @table[id] -%> table <%= @table[id] %><% end %><%- if @metric and @metric[id] -%> metric <%= @metric[id] %><% end %><%- end -%>
```

Comment:
@TorLdre does something bad happens if you place the inet6 route both here and in the route6 file?
If it doesn't harm I'd just leave this template as is, as it's not trivial to fix this without further refactorings.

Reply:
inet6 routes in the route-${name} file creates errors, as inet4 routes creates errors when placed in route6-{name} files. On RedHat based systems, one must place the routes for different families in the two different files. This is, as noted earlier, opposed to debian based systems, where all routes are placed in the same file. I still think my proposal is the easiest way out without changing existing behaviour (for the debian users).

Reply:
About the dangers you refer to, @alvagante, it is true that routes will be removed when "inet6" is removed, but today specifying "inet6" just generates errors on RedHat based systems, so it is an upgrade from today, regardless.

Reply:
@TorLdre last time I checked the problem was that existing ipv4 routes were removed.
Will double check and eventually add some workaround to preserve backwards compatibility

Reply:
@alvagante in the template above ( templates/route-RedHat.erb ) all routes are written _unless_ you specify "inet6" in family. If no family parameter is provided, the route is written to route-${name}. In the route6-${name} template a route is written if, and only if, the family parameter matches "inet6". I can't see how this breaks backward compatibility.

Reply:
@TorLdre $family by default is undef, so the check added in the template would remove the line for the routes where family is not specified

Reply:
@alvagante Oh my... You are of course right! My apologies for wasting your time. My logic failed. Anyway, I see two ways around the problem:

1) Put in an extra statement in the template (if family is undef then write route i route-${name} to fix what I intended to do in the first place. I don't like the solution, though.

2) Change default value for "family" from undef to something, for example inet4. This will maintain default behaviour as the in the existing template for debian refers to "if familiy and if family is inet6". This, I think, is less evil than 1).

3) iterate the hash if on RedHat platform and check if family is set and equals inet6 on one ore more routes. This may be the best solution (and will not create the extra file if unnecessary), but will require some work.

Would you accept 2)?

Reply:
Go for 2, thx!